plugins/bcollection/_keys.py

```diff
@@ -0,0 +1,445 @@
+                while it and not check(it, model.get_path(it)):
+                    it = model.iter_next(it)
+
+                return (it is not None)
+
+        def check(it, path):
+            """
+                High level word check and children
+                :param it: Gtk.TreeIter
+                :param path: Gtk.TreePath
+                :return: bool
+            """
+            if not check_word(it, path):
+                if model.get_string_from_iter(it) == first_it_string:
+                    raise StopIteration()
+
+                return check_children(it, path)
+            else:
+                return True
+
+        treated = False
+        it = first_it
+        if len(word) > 1:
+            # Treats first current/initial path
+            treated = (check_word(first_it, first_path))
+
+        if not treated:
+            treated = check_children(first_it, first_path)
+
+        while not treated:
+            it_next = model.iter_next(it)
+            while it_next is None:
+                it_parent = model.iter_parent(it)
+                if it_parent:
+                    it = it_parent
+                    it_next = model.iter_next(it_parent)
+                else:
+                    it_next = model.get_iter_first()
+                    break  # it_parent should never be None here (fallback)
+
+            it = it_next
+            path = model.get_path(it)
+            try:
+                treated = check(it, path)
+            except StopIteration:
+                return False
+
+        return treated
+
+    treated = False
+    if not (evt.state & Gtk.accelerator_get_default_mod_mask()):
+        from datetime import timedelta, datetime
+        old_keyboard_event = tree.last_keyboard_event
+        current_datetime = datetime.utcnow()
+
+        if (old_keyboard_event is None or
+                (old_keyboard_event + timedelta(seconds=2)) <= current_datetime):
+            tree.word = ''
+
+        keyval_unicode = Gdk.keyval_to_unicode(evt.keyval)
+        if keyval_unicode:
+            keyval_char = unichr(keyval_unicode)
+            word = tree.word + keyval_char
+
+            try:
+                treated = treat_word(word)
+            except StopIteration:
+                pass
+
+            if not treated and len(word) > 1:
+                word = unicode(keyval_char)
+                try:
+                    treated = treat_word(word)
+                except StopIteration:
+                    pass
+
+            if treated:
+                tree.word = word
+                tree.last_keyboard_event = current_datetime
+
+    return treated
+
+
+def key_pressed_to_change_font_size(tree, evt):
+    """
+        Check if key pressed is to change font size (Ctrl +/-)
+        :param tree: Gtk.TreeView
+        :param evt: Gdk.EventKey
+        :return: bool
+    """
+    font_size_factor = 0
+    if evt.state & Gtk.accelerator_get_default_mod_mask():
+        if evt.keyval == Gdk.KEY_KP_Add:
```

Comment:
But I'm not considering this keys purposely, because + and - are valid keys to surf in tree, so, if the user want to change font size and do not have the num pad, he will need to change it on preferences.